.github/workflows/pdp_cicd.yml

```diff
@@ -0,0 +1,102 @@
+name: Build and Push PDP Docker Image
+on:
+  release:
+    types: [published]
+jobs:
+  build-and-push-pdp-vanilla:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Pre build - for PDP-Vanilla
+      run: echo "${{ github.event.release.tag_name }}" | cut -d '-' -f 1 > permit_pdp_version
```

Comment:
If you anyway gonna have some makefile for that, perhaps you can reuse the makefile

Reply:
The make file would be for local use